### ISSUE TITLE
[Feature Request] Add audio files from Wiktionaries into .zim files

### ISSUE THREAD

**Issue by kelson42** [openzim/mwoffliner#1652]
From overview created by [ghost](https://github.com/ghost): kiwix/overview#60  

@kelson42     Some Wiktionaries contain a huge number of pronunciations in .ogg format by native speakers. As an example, the German Wiktionary contains around 700,000 pronunciations.

The current .zim files of Wiktionaries _usually_ contain only plain text and do not include the audio files.

Language learners in developing countries with no internet connection cannot access online websites to listen pronunciations. Wiktionaries include IPA transcriptions, but it is not enough. The actual pronunciation of a native speaker is very helpful.

**Would it be possible to include pronunciations from Wiktionaries in .zim files offered by Kiwix** ?

EDIT:  _- A method to reduce the file size of .ogg format audios would be converting them into .opus format. That format conversion can reduce the size of audios by  60-70%.

Here is a screenshot of a German Wiktionary .zim file used on [GOLDENDICT](https://en.wikipedia.org/wiki/GoldenDict) :

![image](https://user-images.githubusercontent.com/58692066/144313187-2c5e4857-50c6-48da-9419-41c1712e5904.png)

![image](https://user-images.githubusercontent.com/58692066/143959088-2984d8b8-7ec6-4766-9188-f5049aa4f663.png)
------------------------
_PS.  1) I use the German Wiktionary in .zim format with [GoldenDict](https://en.wikipedia.org/wiki/GoldenDict). I live in a remote village in South America. Thank you very much, really ! The Kiwix Project has saved me because I almost never have internet connection._

2)  The German Wiktionary currently weighs 1.4 GiB. If pronunciations in .opus format are added, it would weight around 5-6 GiB.

3) The Wiktionaries with more audios are English, French, and German.  If compression is used in audio files, the tradeoff would be reasonable. At most 3-4 GiB would be added to Wiktionaries in the main languages.  Other languages as Spanish and Italian have fewer pronunciations, and it would be less than 1GiB to be added to .zim files.

**Comment by kelson42** [comment]
Some Wiktionaries (e.g. French) might have >10 pronunciations per word. For example the word "[maison](https://fr.wiktionary.org/wiki/maison)" has 13 audio files. 

As far as I know, only the French Wiktionary has so many audios because they have a separate section for Pronunciation and add audios in bulk using @lingua-libre  [ www.lingualibre.org ]. Suggestions to reduce the size of .zim files with audios:
- Limit the number of .ogg files per entry (e.g. max. 3)
- convert .ogg file to another format for compression (e.g., .opus)

![image](https://user-images.githubusercontent.com/58692066/144400448-f056e034-b68a-42bc-a18e-d96ef83b38f9.png)

**Comment by kelson42** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be now be reviewed manually. Thank you for your contributions.

**Comment by kelson42** [comment]
dafuq is wrong with these people. I'm francophone and can maybe pick three differences within these twelve.
We need to parse and pick up one sound, otherwise we'll end up with tons of junk like the above.

**Comment by kelson42** [comment]
Some automatic tools can be used to evaluate the pronounciation.

**Comment by kelson42** [comment]
We can do so, mwoffliner can do it... but this is a bit subptil. We want the `ogg` but not the video files for example. We should have a look in detail how tod o.

**Comment by kelson42** [comment]
This makes a lot of sense, at least in a Wiktionary context. I'm not super convinced that adding yet another flavour to the existing three (mini/nopic/maxi) would do us any good.
The question at this stage would therefore be: could we tweak mwoffliner so that it produces wiktionary files that include the .ogg files in maxi?

**Comment by Jaifroid** [comment]
@kelson42 Does Wiktionary actually have lots of video files? My impression is that video is very rare, if used at all. So in that case, you could just try an 'all'/full scrape (not maxi) on a smaller dictionary like the Spanish one for testing the file size increase, assuming that mwOffliner can pick up .ogg audio when it's in full mode. Wikitionaries are not very big ZIM files anyway, compared to Wikipedia...

**Comment by kelson42** [comment]
> @kelson42 Does Wiktionary actually have lots of video files?

Yes, but a few audio files. Actually for some reason it seems we have already a few audio files. I need to assess the situation.

**Comment by Immunize2** [comment]
There are 3 Wiktionaries with planty of audio files:
- English
- German
- French

The German version has 800.000 audios approximately. Around 98% of audios are in .ogg format but there are also .wav files. 

Videos are so rare that they can be ignored. However, pronunciations from Wiktionary are extremely useful for language learners with no internet connection. 

_Just imagine, how can someone learn to pronounce English words without being able to listen them ?_ 

_This issue is quite important for young students learning languages and the size of audio files is not so big in comparison with the Wikipedia._

**Comment by Jaifroid** [comment]
Given the [query in Reddit](https://www.reddit.com/r/Kiwix/comments/1iy21ke/question/), has anyone tried to see if it might "just work" to do a full (not maxi) scrape of a smaller Wiktionary, like Spanish? Wouldn't that be pretty simple to try (maybe using the Desktop API, if the Mobile one is still causing issues).

**Comment by audiodude** [comment]
> Given the [query in Reddit](https://www.reddit.com/r/Kiwix/comments/1iy21ke/question/), has anyone tried to see if it might "just work" to do a full (not maxi) scrape of a smaller Wiktionary, like Spanish? Wouldn't that be pretty simple to try (maybe using the Desktop API, if the Mobile one is still causing issues).

Assuming we "just stick the audio files in the ZIM" how does the ZIM reader go about playing them back?

**Comment by benoit74** [comment]
I've just requested a full, a maxi and a nopic of wiktionary ES with WikimediaDesktop API, let's see: https://farm.openzim.org/pipeline/83682772-a75c-4d46-9413-5181aaba1d66

**Comment by Jaifroid** [comment]
> Assuming we "just stick the audio files in the ZIM" how does the ZIM reader go about playing them back?

If the browser can play back the files on the original site, it ought to be able to play them back from the ZIM. In most cases the back end acts just like a server and transparently serves the file(s) and any JS requested by the browser. So long as paths have been correctly rewritten by mwOffliner. Also applies to the JS apps, where the Service Worker does the exact same thing. This might not work in all Web views, but we can only test that empirically, I think.

**Comment by benoit74** [comment]
Wiktionary ES was probably not the best idea, it looks like there is rarely pronunciation. At least I found one word with pronunciation but it is missing in the (full) ZIM : https://es.wiktionary.org/wiki/cuchara / https://dev.library.kiwix.org/viewer#wiktionary_es_all_2025-02/A/cuchara

Will need to dig deeper to understand the situation

**Comment by Jaifroid** [comment]
I found another one: the article for "whisky". This has a pronunciation (of the English) under https://es.wiktionary.org/wiki/whisky, but it's missing in https://dev.library.kiwix.org/viewer#wiktionary_es_all_2025-02/A/whisky .

At least we've discovered that mwOffliner is not fetching these assets, and that needs to be unblocked at least for "full" ZIMs.
